applications/serial_lte_modem/overlay-carrier.conf

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+CONFIG_LWM2M_CARRIER=y
+
+# Workaround for known issue NCSDK-16916.
+# Set the size of the storage partition to a multiple of CONFIG_NRF_SPU_FLASH_REGION_SIZE.
+CONFIG_LWM2M_CARRIER_STORAGE_AREA_SIZE=0x8000
+
+# LwM2M carrier is only compiled for hard-float
+CONFIG_FP_HARDABI=y
+
+# PDN management support
+CONFIG_PDN=y
+
+# Download client for DFU
+CONFIG_DOWNLOAD_CLIENT_MAX_FILENAME_SIZE=230
+
+# Modem info
+CONFIG_MODEM_INFO_BUFFER_SIZE=512
+
+# Disable AT host library as it's incompatible with the carrier library's
+# requirement to initialize the modem library
+CONFIG_AT_HOST_LIBRARY=n
+
+# Credential management
+CONFIG_MODEM_KEY_MGMT=y
+
+# Non-volatile Storage
+CONFIG_NVS=y
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
+
+# Default partition for NVS is unused
+CONFIG_PM_PARTITION_SIZE_NVS_STORAGE=0
+
+# Enable LwM2M carrier support in SLM
+CONFIG_SLM_CARRIER=y
+
+# Disable GNSS to have enough space for carrier library
+CONFIG_SLM_GNSS=n
+CONFIG_NRF_CLOUD_PGPS=n
+
+# LwM2M carrier currently does not support reinitialization
+CONFIG_NRF_MODEM_LIB_ON_FAULT_DO_NOTHING=y
```

Comment:
LwM2M carrier library currently does not support reinitialization without reboot.This will be solved in the future (please refer to NCSDK-17438).